### ISSUE TITLE
Airspy Mini setting sample rate

### ISSUE THREAD

**Issue by john-** [madengr/ham2mon#25]
I am attempting to use an Airspy Mini with sample rate of 3 MS/s (it supports that and 6 MS/s).   I am starting ham2mon like this:

ham2mon.py -a "airspy" -r 3E6 -w -m -b 16

However, the gui shows 10 MS/s and not 3.   I haven't gotten into the code yet and want to make sure I am not missing something basic.

**Comment by madengr** [comment]
Try just running receiver.py

Look at the code at the bottom of that file.  There is an ask_samp_rate which will be 3E6 for you, but the true sample rate is what the osmocom driver returns.  May be an issue with the driver.  I don’t have a mini, but the original airspy only did 10E6, and I think 2.5E6.

**Comment by john-** [comment]
Thanks!   I put a print statement in there that confirmed driver always returns 10 MS/s no matter what I pass it:

        self.samp_rate = self.src.get_sample_rate()
        print "samp_rate %s : %.3f Msps" % (hw_args, self.samp_rate/1E6)

After this I tried the Airspy Mini on my Fedora laptop and it works (sets requested sample rate).  It is not working on the Odroid C2 (device I reported the problem on).

I will close this ticket and investigate the driver issue.   The rest is for "the next person"...

I do see the OS (dmesg) reports the Airspy differently on the Fedora laptop vs. Odroid C2.

Fedora:
[94004.600016] usb 2-1: new high-speed USB device number 24 using xhci_hcd
[94004.726858] usb 2-1: New USB device found, idVendor=1d50, idProduct=60a1, bcdDevice= 1.00
[94004.726864] usb 2-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[94004.726867] usb 2-1: Product: AIRSPY
[94004.726870] usb 2-1: Manufacturer: www.airspy.com
[94004.726872] usb 2-1: SerialNumber: AIRSPY SN:<snip>

C2:
[ 1915.526275] usb 1-1.1: USB disconnect, device number 3
[ 2291.672954] usb 1-1.1: new high-speed USB device number 5 using dwc_otg

airspy_info reports the same on both:

Fedora:
airspy_lib_version: 1.0.9

Found AirSpy board 1
Board ID Number: 0 (AIRSPY)
Firmware Version: AirSpy MINI v1.0.0-rc10-3-g7120e77 2018-04-28
Part ID Number: 0x6906002B 0x00000030
Serial Number: <snip>
Supported sample rates:
        6.000000 MSPS
        3.000000 MSPS
Close board 1

C2:

airspy_lib_version: 1.0.9

Found AirSpy board 1
Board ID Number: 0 (AIRSPY)
Firmware Version: AirSpy MINI v1.0.0-rc10-3-g7120e77 2018-04-28
Part ID Number: 0x6906002B 0x00000030
Serial Number: <snip>
Supported sample rates:
        6.000000 MSPS
        3.000000 MSPS
Close board 1
